src/clparser.cc

```diff
@@ -45,8 +45,8 @@ string CLParser::FilterShowIncludes(const string& line,
   const char* in = line.c_str();
   const char* end = in + line.size();
   const string& prefix = deps_prefix.empty() ? kDepsPrefixEnglish : deps_prefix;
-  if (end - in > (int)prefix.size() &&
-      memcmp(in, prefix.c_str(), (int)prefix.size()) == 0) {
+  if (end - in > static_cast<int>(prefix.size()) &&
```

Comment:
This is not the right thing to do.

decltype(pointer - pointer) == std::ptrdiff_t.

Put parenthesise around (end - in) to avoid operator precedence issues.

Reply:
> Put parenthesise around (end - in) to avoid operator precedence issues.

That would just be a readability improvement (debatable), as the operator precedence is well defined.

Reply:
I've seen enough situations nowhere people have screwed it up that I'd say better safe than sorry.

Reply:
how can you screw up mixing `+` and `>`?

Reply:
I had a boss who messed with parenthesis around `-`. What now?

Reply:
Then we're at an impasse, and you're going to use your authority as the maintainer of the project to make a decision, and I'll put parenthesis around things in my fork.